File: citizen/taxation/taxation.go
package taxation

// Taxation models the citizen tax framework in an American-style system.
type Taxation struct {
	EffectiveIncomeTaxRate float64 // Average effective federal income tax rate (~18.5%)
	CapitalGainsTaxRate    float64 // Long-term capital gains tax rate (~15.0%)
	SalesTaxRate           float64 // Average state & local consumption tax rate (~6.5%)
	TotalIncomeTaxesPaid   float64 // Cumulative USD collected from personal income
	TotalCapGainsTaxesPaid float64 // Cumulative USD collected from capital gains
	TotalSalesTaxesPaid    float64 // Cumulative USD collected from sales
}

// NewDefaultTaxation initializes standard federal and state citizen tax rates.
func NewDefaultTaxation() *Taxation {
	return &Taxation{
		EffectiveIncomeTaxRate: 0.185,
		CapitalGainsTaxRate:    0.150,
		SalesTaxRate:           0.065,
	}
}

// ComputeIncomeTax calculates the tax owed on citizen gross earnings.
func (t *Taxation) ComputeIncomeTax(grossIncome float64) float64 {
	return grossIncome * t.EffectiveIncomeTaxRate
}

// ComputeSalesTax calculates consumption tax on retail spending.
func (t *Taxation) ComputeSalesTax(consumerSpending float64) float64 {
	return consumerSpending * t.SalesTaxRate
}

// ComputeCapitalGainsTax calculates tax on realized market profits.
func (t *Taxation) ComputeCapitalGainsTax(realizedGains float64) float64 {
	if realizedGains <= 0 {
		return 0.0
	}
	return realizedGains * t.CapitalGainsTaxRate
}

// RecordTaxes adds collected receipts to cumulative tallies.
func (t *Taxation) RecordTaxes(incomeTax, capGainsTax, salesTax float64) {
	t.TotalIncomeTaxesPaid += incomeTax
	t.TotalCapGainsTaxesPaid += capGainsTax
	t.TotalSalesTaxesPaid += salesTax
}

// TotalCollected returns total cumulative citizen taxes paid to government.
func (t *Taxation) TotalCollected() float64 {
	return t.TotalIncomeTaxesPaid + t.TotalCapGainsTaxesPaid + t.TotalSalesTaxesPaid
}

File: src/citizen/taxation.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct Taxation {
    pub effective_income_tax_rate: f64,
    pub capital_gains_tax_rate: f64,
    pub sales_tax_rate: f64,
    pub total_income_taxes_paid: f64,
    pub total_cap_gains_taxes_paid: f64,
    pub total_sales_taxes_paid: f64,
}

impl Default for Taxation {
    fn default() -> Self {
        Self {
            effective_income_tax_rate: 0.185,
            capital_gains_tax_rate: 0.150,
            sales_tax_rate: 0.065,
            total_income_taxes_paid: 0.0,
            total_cap_gains_taxes_paid: 0.0,
            total_sales_taxes_paid: 0.0,
        }
    }
}

impl Taxation {
    pub fn new() -> Self {
        Self::default()
    }

    pub fn compute_income_tax(&self, gross_income: f64) -> f64 {
        gross_income * self.effective_income_tax_rate
    }

    pub fn compute_sales_tax(&self, consumer_spending: f64) -> f64 {
        consumer_spending * self.sales_tax_rate
    }

    pub fn compute_capital_gains_tax(&self, realized_gains: f64) -> f64 {
        if realized_gains <= 0.0 {
            0.0
        } else {
            realized_gains * self.capital_gains_tax_rate
        }
    }

    pub fn record_taxes(&mut self, income_tax: f64, cap_gains_tax: f64, sales_tax: f64) {
        self.total_income_taxes_paid += income_tax;
        self.total_cap_gains_taxes_paid += cap_gains_tax;
        self.total_sales_taxes_paid += sales_tax;
    }

    pub fn total_collected(&self) -> f64 {
        self.total_income_taxes_paid + self.total_cap_gains_taxes_paid + self.total_sales_taxes_paid
    }
}
